datadog_checks_dev/datadog_checks/dev/tooling/commands/validate/saved_views.py

```diff
@@ -0,0 +1,147 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import json
+
+import click
+
+from ...utils import complete_valid_checks, get_saved_views, get_valid_integrations, load_saved_views
+from ..console import CONTEXT_SETTINGS, abort, echo_failure, echo_success
+
+REQUIRED_HEADERS = {'name', 'page', 'query', 'type'}
+
+OPTIONAL_HEADERS = {'options', 'timerange', 'visible_facets'}
+
+ALL_HEADERS = REQUIRED_HEADERS | OPTIONAL_HEADERS
+
+VALID_TYPES = {'logs', 'trace'}
+
+VALID_PAGES = {'analytics', 'insights', 'patterns', 'stream', 'traces'}
+
+NO_OPTIONS_PAGES = {'insights', 'patterns', 'traces'}
+
+STREAM_OPTIONS = {
+    "columns",
+    "message_display",
+    "show_date_column",
+    "show_message_column",
+    "show_timeline",
+    "sort",
+    "stream",
+}
+
+ANALYTICS_OPTIONS = {"aggregations", "group_bys", "limit", "order", "step_ms", "widget"}
+
+
+@click.command(context_settings=CONTEXT_SETTINGS, short_help='Validate saved view files')
+@click.argument('integration', autocompletion=complete_valid_checks, required=False)
+def saved_views(integration):
+    """Validates saved view files
+
+    If `check` is specified, only the check will be validated,
+    otherwise all saved views files in the repo will be.
+    """
+    errors = False
+    all_saved_views = {}
+    if integration:
+        all_saved_views[integration] = get_saved_views(integration)
+    else:
+        integrations = sorted(get_valid_integrations())
+        all_saved_views = {}
+        for integration in integrations:
+            all_saved_views[integration] = get_saved_views(integration)
+
+    for integration, saved_views in all_saved_views.items():
+
+        for saved_view in saved_views:
+
+            # all saved views must be json
+            try:
+                view = load_saved_views(saved_view)
+            except json.JSONDecodeError as e:
+                errors = True
+                echo_failure(f"{integration} saved view is not valid json: {e}.")
```

Comment:
May be multiple lines

```suggestion
                echo_failure(f"{integration} saved view is not valid json: {e}")
```